src/utils.js

```diff
@@ -78,9 +78,18 @@ var regexEscape = str => str ? str.replace(/[.*+\-?^${}()|[\]\\]/g, '\\$&') : ''
 var readJSON = file => {
     try {
         const [success, data, /*tag*/] = file.load_contents(null)
-        if (success) return JSON.parse(data instanceof Uint8Array
-            ? ByteArray.toString(data) : data.toString())
-        else throw new Error()
+        if (success) {
+            var raw = data instanceof Uint8Array ? ByteArray.toString(data) : data.toString()
+            if (raw.includes('json-db')) {// quick check
+                try {
+                    var found = raw.match(/<script id="json-db" type="application\/json">(.*)<\/script>/)[1]
```

Comment:
I don't really like how it looks for `/<script id="json-db" type="application\/json">(.*)<\/script>/`. I would probably prefer properly parsing the HTML. But, even if one doesn't want to do that, one still needs to first distinguish between JSON and HTML files, otherwise it would fail if someone were to include `<script id="json-db" type="application/json"></script>` in their notes.

Reply:
Agreed